### PR TITLE
Use fetch in FormData guide

### PR DESCRIPTION
**Pull request by wbamberg**
Part of https://github.com/mdn/content/issues/30131.

The only real weirdness in here is that I removed the link to the Glitch demo, because:
- it used XHR, and I couldn't update it because it's Glitch
- I couldn't even see the code, so it didn't seem particularly helpful

### CODE REVIEW

File: files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md [hunk 1]
```diff
@@ -6,41 +6,46 @@ page-type: guide
...
+  const response = await fetch("http://example.org/post", {
+    method: "POST",
+    body: formData,
+  });
+  console.log(await response.json());
+});
 ```
 
-> **Note:** The fields "userfile" and "webmasterfile" both contain a file. The number assigned to the field "accountnum" is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string: **if the value is neither a `Blob` nor a `File`, the value is converted to a string**).
+> **Note:** The fields "avatar" and "webmasterfile" both contain a file. The number assigned to the field "accountnum" is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string: **if the value is neither a `Blob` nor a `File`, the value is converted to a string**).
```
Comment (teoli2003) [inline files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:40]: Nit:
```suggestion
> **Note:** The fields `"avatar"` and `"webmasterfile"? both contain a file. The number assigned to the field `"accountnum"` is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string: _if the value is neither a `Blob` nor a `File`, the value is converted to a string_).
```

File: files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md [hunk 2]
```diff
@@ -6,41 +6,46 @@ page-type: guide
...
+This example builds a `FormData` instance containing values for fields named "username", "accountnum", "avatar" and "webmasterfile", then uses {{domxref("fetch()")}} to send the form's data. The field "webmasterfile" is a {{domxref("Blob")}}. A `Blob` object represents a file-like object of immutable, raw data. Blobs represent data that isn't necessarily in a JavaScript-native format. The {{ domxref("File") }} interface is based on `Blob`, inheriting blob functionality and expanding it to support files on the user's system. In order to build a `Blob` you can invoke [the `Blob()` constructor](/en-US/docs/Web/API/Blob/Blob).
```
Comment (teoli2003) [inline files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:42]:
```suggestion
This example builds a `FormData` instance containing values for fields named `"username"`, `"accountnum"`, `"avatar"` and `"webmasterfile"`, then uses {{domxref("fetch()")}} to send the form's data. The field `"webmasterfile"` is a {{domxref("Blob")}}. A `Blob` object represents a file-like object of immutable, raw data. Blobs represent data that isn't necessarily in a JavaScript-native format. The {{ domxref("File") }} interface is based on `Blob`, inheriting blob functionality and expanding it to support files on the user's system. In order to build a `Blob` you can invoke [the `Blob()` constructor](/en-US/docs/Web/API/Blob/Blob).
```

File: files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md [hunk 3]
```diff
@@ -6,41 +6,46 @@ page-type: guide
...
-request.open("POST", "https://example.com/submitform.php");
-request.send(formData);
+const send = document.querySelector("#send");
+
+send.addEventListener("click", async () => {
+  const formData = new FormData();
+  formData.append("username", "Groucho");
+  formData.append("accountnum", 123456);
+
+  // file <input> element
```
Comment (teoli2003) [inline files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:23]: NIT: To be consistent with the snippet below (and I like comments starting with a capital letter when at the beginning of a line.

```suggestion
  // A file <input> element
```

File: files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md [hunk 4]
```diff
@@ -182,10 +192,11 @@ formElem.addEventListener("formdata", (e) => {
     console.log(value);
   }
 
-  // submit the data via XHR
-  const request = new XMLHttpRequest();
-  request.open("POST", "/formHandler");
-  request.send(data);
+  // submit the data via fetch()
```
Comment (teoli2003) [inline files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:195]: NIT:
```suggestion
  // Submit the data via fetch()
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
This is nice!

I have a few nits but besides these, this is ready.

File: files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md [hunk 5]
```diff
@@ -6,41 +6,46 @@ page-type: guide
...
+  const response = await fetch("http://example.org/post", {
+    method: "POST",
+    body: formData,
+  });
+  console.log(await response.json());
+});
 ```
 
-> **Note:** The fields "userfile" and "webmasterfile" both contain a file. The number assigned to the field "accountnum" is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string: **if the value is neither a `Blob` nor a `File`, the value is converted to a string**).
+> **Note:** The fields `"avatar"` and `"webmasterfile"? both contain a file. The number assigned to the field `"accountnum"` is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string: _if the value is neither a `Blob` nor a `File`, the value is converted to a string_).
```
[the remark this answers is not in the document]
Comment (wbamberg) [reply files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:40]: I'm not sure why it wants to do this.

Comment (wbamberg) [inline files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:40]:
```suggestion
> **Note:** The fields `"avatar"` and `"webmasterfile"? both contain a file. The number assigned to the field `"accountnum"` is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string. If the value is neither a `Blob` nor a `File`, the value is converted to a string).
```

tbh I don't think this needs any emphasis, I would just make it a separate sentence.

File: files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md [hunk 6]
```diff
@@ -6,41 +6,46 @@ page-type: guide
...
+  const response = await fetch("http://example.org/post", {
+    method: "POST",
+    body: formData,
+  });
+  console.log(await response.json());
+});
 ```
 
-> **Note:** The fields "userfile" and "webmasterfile" both contain a file. The number assigned to the field "accountnum" is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string: **if the value is neither a `Blob` nor a `File`, the value is converted to a string**).
+> **Note:** The fields `"avatar"` and `"webmasterfile"? both contain a file. The number assigned to the field `"accountnum"` is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string. If the value is neither a `Blob` nor a `File`, the value is converted to a string).
```
Comment (wbamberg) [inline files/en-us/web/api/xmlhttprequest_api/using_formdata_objects/index.md:40]:
```suggestion
> **Note:** The fields `"avatar"` and `"webmasterfile"` both contain a file. The number assigned to the field `"accountnum"` is immediately converted into a string by the [`FormData.append()`](/en-US/docs/Web/API/FormData/append) method (the field's value can be a {{ domxref("Blob") }}, {{ domxref("File") }}, or a string. If the value is neither a `Blob` nor a `File`, the value is converted to a string).
```

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**